R/get_nowcast_pred_draws.R

```diff
@@ -100,13 +112,37 @@ get_nowcast_pred_draw <- function(point_nowcast_pred_matrix,
 #' nowcast_pred_draws_df
 get_nowcast_pred_draws <- function(point_nowcast_pred_matrix,
                                    disp,
-                                   n_draws = 1000) {
+                                   n_draws = 1000,
+                                   fun_to_aggregate = sum,
+                                   k = 1) {
+  # Define allowed functions
+  allowed_functions <- list(
+    sum = sum
+  )
+
+  # Validate function
+  fun_name <- deparse(substitute(fun_to_aggregate))
+  if (is.name(fun_to_aggregate)) {
+    fun_name <- as.character(fun_to_aggregate)
+  }
+
+  # Check if function is in allowed list
+  if (!identical(fun_to_aggregate, allowed_functions[[fun_name]]) &&
```

Comment:
Note my point about this not being something I need but if we keep it it should be a function due to DRY